state-sync/state-sync-v2/data-streaming-service/src/streaming_service.rs

```diff
@@ -304,18 +339,7 @@ impl<T: AptosDataClient + Send + Clone + 'static> DataStreamingService<T> {
 
     /// Returns the data stream associated with the given `data_stream_id`.
     /// Note: this method assumes the caller has already verified the stream exists.
-    fn get_data_stream(&self, data_stream_id: &DataStreamId) -> &DataStream<T> {
-        self.data_streams.get(data_stream_id).unwrap_or_else(|| {
-            panic!(
-                "Expected a data stream with ID: {:?}, but found None!",
-                data_stream_id
-            )
-        })
-    }
-
-    /// Returns the data stream associated with the given `data_stream_id`.
-    /// Note: this method assumes the caller has already verified the stream exists.
-    fn get_data_stream_mut(&mut self, data_stream_id: &DataStreamId) -> &mut DataStream<T> {
+    fn get_data_stream(&mut self, data_stream_id: &DataStreamId) -> &mut DataStream<T> {
         self.data_streams
             .get_mut(data_stream_id)
             .unwrap_or_else(|| {
```

Comment:
if assumes the caller has verified the stream exists, we should not return `Ok(())` in the `unwrap_or_else` clause, right?

Reply:
Legacy, I'll update for an explicit panic 😄

Reply:
@lightmark, the full code is:

```
    /// Returns the data stream associated with the given `data_stream_id`.
    /// Note: this method assumes the caller has already verified the stream exists.
    fn get_data_stream(&mut self, data_stream_id: &DataStreamId) -> &mut DataStream<T> {
        self.data_streams
            .get_mut(data_stream_id)
            .unwrap_or_else(|| {
                panic!(
                    "Expected a data stream with ID: {:?}, but found None!",
                    data_stream_id
                )
            })
    }
```

What am I missing?

Reply:
I think I got the version before the force-push. NVM